### PR TITLE
Backport Bitcoin PR#9260: Mrs Peacock in The Library with The Candlestick (killed main.{h,cpp})

### PR DESCRIPTION
**Pull request by OlegGirko**
This is backport of Bitcoin PR bitcoin/bitcoin#9260.

This PR completes split of `main.{h,cpp}` into `net_processing.{h,cpp}` and `validation.{h,cpp}`.
The change is quite intrusive, but it's just code move (and changes in multiple modules to adapt to new include file names).

The original PR description is omitted because it's completely irrelevant to this change. Check for yourself if you don't believe me.

### CODE REVIEW

File: src/net_processing.cpp [hunk 1]
```diff
@@ -0,0 +1,2666 @@
...
+#include "masternodeman.h"
+#include "privatesend-client.h"
+#include "privatesend-server.h"
+
+#include <boost/thread.hpp>
+
+using namespace std;
+
+#if defined(NDEBUG)
+# error "Bitcoin cannot be compiled without assertions."
```
Comment (UdjinM6) [inline src/net_processing.cpp]: s/Bitcoin/Dash Core/

File: src/net_processing.cpp [hunk 2]
```diff
@@ -0,0 +1,2666 @@
...
+
+struct COrphanTx {
+    CTransaction tx;
+    NodeId fromPeer;
+};
+map<uint256, COrphanTx> mapOrphanTransactions GUARDED_BY(cs_main);
+map<uint256, set<uint256> > mapOrphanTransactionsByPrev GUARDED_BY(cs_main);;
+void EraseOrphansFor(NodeId peer) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+
+static const uint64_t RANDOMIZER_ID_ADDRESS_RELAY = 0x3cac0035b5866b90ULL; // SHA256("main address relay")[0:8]
```
Comment (UdjinM6) [inline src/net_processing.cpp]: `RANDOMIZER_ID_ADDRESS_RELAY` is not used, we don't have https://github.com/bitcoin/bitcoin/pull/8688 backported (yet?)

File: src/net_processing.cpp [hunk 3]
```diff
@@ -0,0 +1,2666 @@
...
+
+using namespace std;
+
+#if defined(NDEBUG)
+# error "Bitcoin cannot be compiled without assertions."
+#endif
+
+int64_t nTimeBestReceived = 0; // Used only to inform the wallet of when we last received a block
+
+struct IteratorComparator
```
Comment (UdjinM6) [inline src/net_processing.cpp]: `IteratorComparator` is not used, we don't have https://github.com/bitcoin/bitcoin/pull/8179 (yet?)

**Comment by OlegGirko** [comment]
I tried to make sure that content of `net_processing.cpp` is *exactly* the same as content removed from `main.cpp` in the second commit, but few things still slipped through.

I've pushed my branch updated according to inline comments.

**Pull request merged**